src/hosted-buttons/utils.js

```diff
@@ -225,3 +230,86 @@ export const buildHostedButtonOnApprove = ({
     });
   };
 };
+
+export function getFlexDirection({
+  layout,
+}: GetFlexDirectionArgs): GetFlexDirection {
+  return { flexDirection: layout === "horizontal" ? "row" : "column" };
+}
+
+export function getButtonColor(
+  color: Color,
+  fundingSource: FundingSources
+): Color {
+  const colorMap = {
+    gold: {
+      paypal: "gold",
+      venmo: "blue",
+      paylater: "gold",
+    },
+    blue: {
+      paypal: "blue",
+      venmo: "silver",
+      paylater: "blue",
+    },
+    black: {
+      paypal: "black",
+      venmo: "black",
+      paylater: "black",
+    },
+    white: {
+      paypal: "white",
+      venmo: "white",
+      paylater: "white",
+    },
+    silver: {
+      paypal: "silver",
+      venmo: "blue",
+      paylater: "silver",
+    },
+  };
+
+  return colorMap[color][fundingSource];
+}
+
+export function shouldRenderSDKButtons(
+  fundingSources: $ReadOnlyArray<FundingSources>
+): boolean {
+  return Boolean(fundingSources.length);
+}
+
+export function appendButtonContainer({
+  flexDirection,
+  selector,
+}: BuildButtonContainerArgs) {
+  const elm =
+    typeof selector === "string" ? document.querySelector(selector) : selector;
+
+  if (!elm) {
+    throw new Error("PayPal button container selector was not found");
+  }
+
+  // if (selector.charAt(0) !== "#") {
+  //   throw new TypeError("Selector must be referring to an id");
+  // }
+
+  const buttonContainer = document.createElement("div");
+
+  buttonContainer.setAttribute(
+    "style",
+    `display: flex; flex-wrap: nowrap; gap: 16px; flex-direction: ${flexDirection}`
```

Comment:
I noticed that the horizontal layout will take up 100% of the parent container's width but the v1 buttons had a max-width set:

v1:


<img width="1476" alt="Screenshot 2024-04-17 at 12 37 43" src="https://github.com/paypal/paypal-checkout-components/assets/3824954/3e8f628b-5065-4880-a7cf-c7ad5d574c15">

v2: 
<img width="1727" alt="Screenshot 2024-04-17 at 12 33 28" src="https://github.com/paypal/paypal-checkout-components/assets/3824954/3b607f5a-e280-4df2-8402-a7839a406a2d">

I think one quick win could be adding `max-width: 750px` here.

There will be a small difference from v1 because the horizontal buttons will hide the second button if there is not enough room (< 300px) but I could see handling that in the future (if ever)

Reply:
I will definitely add the max-with property. Hopefully NCP team can include these styles on html the provide us in the updated response 🤞